src/proofPointResolver.ts

```diff
@@ -0,0 +1,60 @@
+import { ProofPoint } from "./proofPoint";
+import { ProofPointId, ProofPointIdType } from "./proofPointId";
+import { StorageProvider } from "./storage";
+
+/**
+ * Proof point resolver
+ * An object capable of resolving a @ProofPointId to a @ProofPoint
+ */
+export interface ProofPointResolver {
+  /**
+   * Resolve the given @PRoofPointId
+   * @param id The ID of the proof point to resolve
+   * @returns The @ProofPoint corresponding to the given ID.
+   */
+  resolve(id: ProofPointId): Promise<ProofPoint>;
+}
+
+/**
+ * General proof point resolver
+ * A @ProofPointResolver capable of handling all types of @ProofPointId
+ */
+export class GeneralProofPointResolver {
```

Comment:
Again, this seems somewhat over-engineered at this point if we dont have other systems to resolve with?

Reply:
Removed